docs: add secure mode document (#268)
planetarium/NineChronicles.Headless: Docs/SECURE_MODE.md

# Secure mode

Since [#230], `feat(graphql): provide secret token used in policy`, it has started to provide a mode to turn on authorization.

> The mode makes important types need to authenticate with HTTP `Authorization` header. 

Reviewer: ```suggestion
The mode makes important types require authentication with HTTP `Authorization` header. 
```

## Usage

> To turn on the mode, you must run the headless application with `--graphql-secret-token-path` option, and passing the path to store secure token in local storage.

Reviewer: ```suggestion
To turn on the mode, you must run the headless application with `--graphql-secret-token-path` option, and pass the path for storing secure token in the local storage.
```

> Then, there will be random 40 bytes token encoded by base64 at the path you passed. And you should read the secret token and request with `Authorization: Basic <SECRET_TOKEN>` header.   

Reviewer: ```suggestion
Then, there will be a random 40 bytes token encoded by base64 at the path you passed. Then, you can use the secret token and make a request with `Authorization: Basic <SECRET_TOKEN>` header.   
```

### Example (/w GraphQL Playground)

It assumes you cloned this repository and it will work on the cloned directory.

```
$ docker run -p NineChronicles.Headless -- \
    # other options
    --graphql-server \
    --graphql-host localhost \
    --graphql-port <PORT> \
    --graphql-secret-token-path $PWD/secret-token
```

Then, there will be the secret token at `$PWD/secret-token`.

```
$ cat $PWD/secret-token
Jhlgn875txcwD5rKP38bs8ZbvYxnm3hliqYjNjETH8Bo6aX4Cbf6Ng==
``` 

> Though I attached the secret token, it will be generated randomly for each running headless.

Reviewer: ```suggestion
The secret token above is an example and it will be generated randomly for each running headless.
```

> When you request mutations or some queries to GraphQL endpoint, you must do it with `Authorization: Basic <SECRET_TOKEN>`.

Reviewer: Maybe it's a good idea to write a list of queries and mutations that will require a secret token and update it as we add more queries?

Author: I applied it in a3e34d93f8144295d4c3e05f4544ded54aff5ff1. Could you review again?
